src/systemd.rs

```diff
@@ -2,13 +2,27 @@
 
 use color_eyre::{eyre::WrapErr, Result};
 
+pub async fn set_systemd_environment(key: &str, value: &str) -> Result<()> {
+	let _ = std::process::Command::new("systemctl")
+		.arg("--user")
+		.arg("set-environment")
+		.arg(format!("{key}={value}"))
+		.spawn()
+		.wrap_err("Failed to set environment variable in user service manager")?;
+
+	// always return Ok, making systemd optional
+	Ok(())
+}
+
 pub async fn start_systemd_target() -> Result<()> {
 	let _ = std::process::Command::new("systemctl")
 		.arg("start")
 		.arg("--user")
 		.arg("cosmic-session.target")
 		.spawn()
 		.wrap_err("Failed to start cosmic-session.target")?;
```

Comment:
Given the "?" at the end, I am not sure this always returns`Ok`.
If we don't want to handle failure we should make sure the caller doesn't evaluate the result.

Reply:
Yeah, I totally missed the `?` there

I also just noticed that I used [`spawn()`](https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.spawn) here, which does not wait for the command to finish, which is not good considering the whole point of this PR is to improve the sequence of operations

Reply:
Oh wait, we're using std::process::Command here, not tokio::process::Command here ...